sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -145,13 +143,27 @@ private Handler buildBaseHandler(Class<?> sqlObjectType, Method method) {
                             method.getName(),
                             sqlMethodAnnotations));
         }
+        if (method.isDefault()) {
+            if (!sqlMethodAnnotations.isEmpty()) {
+                throw new IllegalStateException(String.format(
+                        "Method %s.%s has mutually exclusive default implementation and a SQL method annotation %s " +
+                                "(must be one or the other)",
+                        sqlObjectType.getSimpleName(),
+                        method.getName(),
+                        sqlMethodAnnotations.get(0).getName()));
+            }
+            return new DefaultMethodHandler();
+        }
 
         return sqlMethodAnnotations.stream()
                 .map(type -> type.getAnnotation(SqlMethodAnnotation.class))
                 .map(a -> buildFactory(a.value()))
                 .map(factory -> factory.buildHandler(sqlObjectType, method))
                 .findFirst()
-                .orElseGet(PassThroughHandler::new);
+                .orElseThrow(() -> new AbstractMethodError(String.format(
```

Comment:
It feels weird for one error path to throw `IllegalStateException` and another to throw `AbstractMethodError`.  I get why, and I can't think of anything better off the top of my head, but it caught my eye.